mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -613,6 +614,12 @@ struct GridwiseGemmRewritePattern : public OpRewritePattern<GridwiseGemmOp> {
     LLVM_DEBUG(llvm::dbgs() << "a_block_space: " << a_block_space << "\n");
     LLVM_DEBUG(llvm::dbgs() << "b_block_space: " << b_block_space << "\n");
     LLVM_DEBUG(llvm::dbgs() << "double_block_space: " << block_space << "\n\n");
+
+    // TODO: adjust for data type and device
+    if (block_space * sizeof(float) > 64 * 1024)
+      return failure();
+    else
```

Comment:
Linter will yell at me for the else there, we could just make this a 
```
return success();
```

More interestingly, both `success()` and `failure()` take boolean arguments.
So you can `return failure(block_space > ...);` and that'll do the right thing

Reply:
Actually linter is able to tell if there's a return for both if and else statement at the end of the return, so this should be fine.

Agreed though that a last catch all return is a better style. Addressed in f5e59f4

Reply:
My copy of clang-tidy whines about
```
if (blah) {
 ...
  return foo;
} else {
  ...
}
```

Reply:
If else also have a return then it shouldn't... Wired.

Reply:
Or maybe if { ... return } else { ... return } is fine, I haven't looked